### PR TITLE
System module

### PR DESCRIPTION
**Pull request by Starbuck5**
From [https://github.com/pygame/pygame/pull/3656](https://github.com/pygame/pygame/pull/3656)

### CODE REVIEW

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

Tested with:

```
import pygame

print(pygame.system.get_pref_locales())
print(pygame.system.get_pref_path(org="Robotic Shed", app="Foulness"))

```

Everything still worked as it should (this PR was already tested and debated on the old repo). 


**NOTE:** You need to do a  full `python -m buildconfig` setup step for this one because otherwise a `pip install .` will not compile the new module.

File: docs/reST/ref/system.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+| :sl:`pygame module to provide additional context about the system`
+
+**EXPERIMENTAL!** This API may change or disappear in later pygame releases. 
+If you use this, your code may break with the next pygame release.
+This is a new module, so we are marking it experimental for now.
+We probably won't have to change API, but we're keeping the possibility
+open just in case something obvious comes up.
+
+.. versionadded:: 2.1.4
```
Comment (oddbookworm) [inline docs/reST/ref/system.rst:17]: Should we see about a modified `versionadded` to specify that it's a pygame-ce thing?
image

Comment (Starbuck5) [reply docs/reST/ref/system.rst]: Personally I don't feel like adding the additional complexity to make it happen.

File: docs/reST/ref/system.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   .. note::
+        Since the organization and app names can potentially be used as
+        a folder name, it is highly encouraged to avoid punctuation.
+        Instead stick to letters, numbers, and spaces.
+
+   .. note::
+        The ``appdirs`` library has similar functionality for this use case,
+        but has more "folder types" to choose from.
+    
+   .. versionadded:: 2.1.4
```
Comment (MyreMylar) [inline docs/reST/ref/system.rst:58]:
```suggestion
   .. versionadded:: 2.2.0
```

File: src_c/doc/system_doc.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+pygame.system.get_pref_path
+ get_pref_path(org, app) -> path
+get a writeable folder for your app
+
+pygame.system.get_pref_locales
+ get_pref_locales() -> list[locale]
+get preferred locales set on the system
+
+*/
```
Comment (MyreMylar) [inline src_c/doc/system_doc.h:22]: Will change - assuming we merge the new doc line generation PR first.

File: docs/reST/ref/system.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+:mod:`pygame.system`
+======================
+
+.. module:: pygame.system
+    :synopsis: pygame module to provide additional context about the system
+
+| :sl:`pygame module to provide additional context about the system`
+
+**EXPERIMENTAL!** This API may change or disappear in later pygame releases.
```
Comment (oddbookworm) [inline docs/reST/ref/system.rst:11]: Are we keeping this? I thought this was decided against

**Review by oddbookworm — APPROVED** [review]
Other than my comment, it looks good to me. I won't merge it yet though for two reasons:
1) my comment might be cause for another edit
2) this is a big enough thing that more approvals might be warranted, I'm not sure

File: docs/reST/ref/system.rst (hunk 4 above)
Comment (Starbuck5) [reply docs/reST/ref/system.rst:11]: Module level experimental is fine. We decided against adding scary notices to every new function we add.

**Pull request merged**